packages/ord-connect/src/lib/signMessage.ts

```diff
@@ -32,5 +42,17 @@ export default async function signMessage({
     return base64;
   }
 
+  if (wallet === Wallet.LEATHER) {
+    const paymentType =
+      walletAddresses.ordinals === address
+        ? LeatherAddressType.P2TR
+        : LeatherAddressType.P2WPKH;
```

Comment:
use `ADDRESS_FORMAT_TO_TYPE` and a constant `ADDRESS_TYPE_TO_LEATHER_ADDRESS_TYPE` to convert the format into a LeatherAddressType enum (it's the same but a helper would ensure type safety from AddressType -> LeatherAddressType)

Reply:
created a function `leatherPaymentTypeFromFormat` to convert address_format to `leather_address_type`